scripts/base/frameworks/analyzer/dpd.zeek

```diff
@@ -63,9 +67,12 @@ event analyzer_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count)
 	add c$service[analyzer];
 	}
 
-event analyzer_violation(c: connection, atype: AllAnalyzers::Tag, aid: count,
-                         reason: string) &priority=10
+event analyzer_violation_info(atype: AllAnalyzers::Tag, info: AnalyzerViolationInfo) &priority=10
```

Comment:
This keeps limiting DPD to connection analyzers, right? Is the plan to extend this to files later?

Reply:
With DPD you're thinking only dpd.log, or the disable_analyzer() logic?

* I could see a less connection-oriented log for tracking analyzer_violations where the analyzer would be the primary entity of interest (or extend the existing dpd.log with fuid and make connection fields optional).

* About `disable_analyzer()`: If a FileAnalyzer's currently returns `false` from it's `DeliverStream()` method, it's disabled - didn't plan to change something around here right now, though let me know if that's another topic for consideration (motivation might have been the hashes where that's the right thing to do).

https://github.com/zeek/zeek/blob/b6512e82c57f2b632b6b328a3a413d77c55d79c8/src/file_analysis/File.cc#L420-L424

Reply:
I could see the same try-to-parse-and-see-if-it-works logic for files that we do for protocols. That way, we could rely less on signature for MIME types, maybe not at all anymore (not sure). But ok to leave that for later.